ravenframework/SupervisedLearning/MultiResolutionTSA.py

```diff
@@ -0,0 +1,184 @@
+    trainedParams = list(self._globalROM._tsaTrainedParams.items())
+    mrAlgo, mrTrainedParams = trainedParams[-1]
+
+    # eh, maybe this should live in TSAUser in the future...
+    # extract settings used for that last algorithm (should have some sort of "levels")
+    numLevels = mrAlgo._getDecompositionLevels()
+
+    # reformat the trained params
+    sortedTrainedParams = mrAlgo._sortTrainedParamsByLevels(mrTrainedParams)
+    return numLevels, sortedTrainedParams
+
+  def _updateMRTrainedParams(self, params):
+    """
+      Method to update trained parameter dictionary from this class using imported `params` dictionary
+      containing multiple-decomposition-level trainings
+      @ In, params, dict, dictionary of trained parameters from previous decomposition levels
+      @ Out, None
+    """
+    # get all trained parameters from final algorithm (should be multiresolution transformer)
+    trainedParams = list(self._globalROM._tsaTrainedParams.items())
+    mrAlgo, mrTrainedParams = trainedParams[-1]
+
+    mrAlgo._combineTrainedParamsByLevels(mrTrainedParams, params)
+
+
+  def writeXML(self, writeTo, targets=None, skip=None):
+    """
+      Allows the SVE to put whatever it wants into an XML to print to file.
+      Overload in subclasses.
+      @ In, writeTo, xmlUtils.StaticXmlElement, entity to write to
+      @ In, targets, list, optional, unused (kept for compatability)
+      @ In, skip, list, optional, unused (kept for compatability)
+      @ Out, None
+    """
+    return
+
+  def __evaluateLocal__(self, featureVals):
+    """
+    Evaluate algorithms for ROM generation
+      @ In, featureVals, float, a scalar feature value is passed as scaling factor
+      @ Out, rlz, dict, realization dictionary of values for each target
+    """
+    rlz = self._globalROM.evaluateTSASequential()
+    return rlz
+
+
+
+
+
+  ### ESSENTIALLY UNUSED ###
+  def _localNormalizeData(self,values,names,feat):
+    """
+      Overwrites default normalization procedure, since we do not desire normalization in this implementation.
+      @ In, values, unused
+      @ In, names, unused
+      @ In, feat, feature to normalize
+      @ Out, None
+    """
+    self.muAndSigmaFeatures[feat] = (0.0,1.0)
+
+  def __confidenceLocal__(self,featureVals):
+    """
+      This method is currently not needed for ARMA
+    """
+    pass
+
+  def __resetLocal__(self,featureVals):
+    """
+      After this method the ROM should be described only by the initial parameter settings
+      Currently not implemented for ARMA
+    """
+    pass
+
+  def __returnInitialParametersLocal__(self):
+    """
+      there are no possible default parameters to report
+    """
+    localInitParam = {}
+    return localInitParam
+
+  def __returnCurrentSettingLocal__(self):
+    """
+      override this method to pass the set of parameters of the ROM that can change during simulation
+      Currently not implemented for ARMA
+    """
+    pass
```

Comment:
Do you need to keep these methods? I assume they are also defined in the base class.

Reply:
it seems we didn't need `_localNormalizeData` but the others are abstract methods in `SupervisedLearning` so they need to be defined here